Index: Source_Files/Sound/VorbisDecoder.h
===================================================================
--- Source_Files/Sound/VorbisDecoder.h	(revision 4513)
+++ Source_Files/Sound/VorbisDecoder.h	(working copy)
@@ -27,8 +27,13 @@
 #include "cseries.h"
 #include "Decoder.h"
 #ifdef HAVE_VORBISFILE
-#include <vorbis/vorbisfile.h>
 
+#if defined(__WII__)
+# include <tremor/ivorbisfile.h>
+#else
+# include <vorbis/vorbisfile.h>
+#endif
+
 class VorbisDecoder : public StreamDecoder
 {
 public:
Index: Source_Files/Sound/VorbisDecoder.cpp
===================================================================
--- Source_Files/Sound/VorbisDecoder.cpp	(revision 4513)
+++ Source_Files/Sound/VorbisDecoder.cpp	(working copy)
@@ -94,7 +94,11 @@
 	int current_section = 0;
 	while (total_bytes_read < max_length)
 	{
+#if defined(__WII__)
+		bytes_read = ov_read(&ov_file, (char *) &buffer[total_bytes_read], max_length - total_bytes_read, &current_section);
+#else
 		bytes_read = ov_read(&ov_file, (char *) &buffer[total_bytes_read], max_length - total_bytes_read, IsLittleEndian() ? 0 : 1, 2, 1, &current_section);
+#endif
 		if (bytes_read <= 0)
 			return total_bytes_read;
 		total_bytes_read += bytes_read;
