Index: SDL_ttf/Makefile
===================================================================
--- SDL_ttf/Makefile	(revision 99)
+++ SDL_ttf/Makefile	(working copy)
@@ -80,7 +80,7 @@
 
 # Compilation flags.
 COMMON_FLAGS	:= -g -O3 -mrvl -Wall $(MACHDEP)
-INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I$(LIBOGC_INC)/SDL -I$(LIBOGC_INC)/freetype
+INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I$(INSTALL_HEADER_DIR)/SDL -I$(LIBOGC_INC)/freetype
 DEFINES			:= -DGEKKO -DDEBUG_ERROR -DDEBUG_TIMERS -DDEBUG_THREADS -DDEBUG_BUILD -DDEBUG_CONVERT 
 CFLAGS			:= -L$(PORTLIBS)/lib -I$(PORTLIBS)/include -I$(PORTLIBS)/include/freetype2 $(COMMON_FLAGS) $(INCLUDES) $(DEFINES)
 
Index: SDL/Makefile
===================================================================
--- SDL/Makefile	(revision 99)
+++ SDL/Makefile	(working copy)
@@ -49,8 +49,10 @@
 # Library object files.
 SDL_OBJS	:= $(subst $(SDL_SRC_DIR),$(SDL_OBJ_DIR),$(SDL_SRCS:.c=.o))
 
+SDLMAIN_OBJS	:= $(SDL_OBJ_DIR)/main/wii/SDL_wii_main.o
+
 # What's a full build?
-all: $(LIB_DIR)/libSDL.a
+all: $(LIB_DIR)/libSDL.a $(LIB_DIR)/libSDLmain.a
 
 # How to delete the intermediate files.
 clean:
@@ -79,6 +81,10 @@
 	powerpc-eabi-ar crs $@ $(SDL_OBJS)
 	@echo ----
 
+$(LIB_DIR)/libSDLmain.a): $(SDLMAIN_OBJS)
+	$(AR) cru $@ $(SDLMAIN_OBJS)
+	$(RANLIB) $@
+
 # How to build a DOL.
 $(BIN_DIR)/%.dol: $(TEST_OBJ_DIR)/%.elf
 	@echo Creating DOL $@
Index: SDL_image/Makefile
===================================================================
--- SDL_image/Makefile	(revision 99)
+++ SDL_image/Makefile	(working copy)
@@ -80,7 +80,7 @@
 
 # Compilation flags.
 COMMON_FLAGS	:= -g -O3 -mrvl -Wall $(MACHDEP)
-INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I$(LIBOGC_INC)/SDL -I$(LIBOGC_INC)/jpeg
+INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I$(INSTALL_HEADER_DIR)/SDL -I$(LIBOGC_INC)/jpeg
 # -I$(DEVKITPRO)/libpng/include
 DEFINES			:= -DGEKKO -DDEBUG_ERROR -DDEBUG_TIMERS -DDEBUG_THREADS -DDEBUG_BUILD -DDEBUG_CONVERT -DLOAD_BMP -DLOAD_JPG -DLOAD_GIF -DLOAD_LBM -DLOAD_PCX -DLOAD_PNM -DLOAD_TGA -DLOAD_XCF -DLOAD_XPM -DLOAD_XV -DLOAD_PNG
 # 
Index: SDL_mixer/Makefile
===================================================================
--- SDL_mixer/Makefile	(revision 99)
+++ SDL_mixer/Makefile	(working copy)
@@ -81,7 +81,7 @@
 
 # Compilation flags.
 COMMON_FLAGS	:= -g -O3 -mrvl -Wall $(MACHDEP)
-INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I$(LIBOGC_INC)/SDL -I$(LIBOGC_INC)/mpeg -I$(DEVKITPRO)/libsmpeg/include -Isrc/mikmod -Isrc/timidity
+INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I$(INSTALL_HEADER_DIR)/SDL -I$(INSTALL_HEADER_DIR)/smpeg -I$(DEVKITPRO)/libsmpeg/include -Isrc/mikmod -Isrc/timidity
 # -include sys/wait.h 
 # -include sys/syslimits.h
 # -Isrc/native_midi
Index: SDL_gfx/Makefile
===================================================================
--- SDL_gfx/Makefile	(revision 99)
+++ SDL_gfx/Makefile	(working copy)
@@ -58,7 +58,7 @@
 
 # Compilation flags.
 COMMON_FLAGS	:= -g -O3 -mrvl -Wall $(MACHDEP)
-INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I$(LIBOGC_INC)/SDL
+INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I$(INSTALL_HEADER_DIR)/SDL
 DEFINES			:= -DGEKKO -DDEBUG_ERROR -DDEBUG_TIMERS -DDEBUG_THREADS -DDEBUG_BUILD -DDEBUG_CONVERT
 CFLAGS			:= $(COMMON_FLAGS) $(INCLUDES) $(DEFINES)
 
Index: SDL_net/Makefile
===================================================================
--- SDL_net/Makefile	(revision 99)
+++ SDL_net/Makefile	(working copy)
@@ -56,7 +56,7 @@
 
 # Compilation flags.
 COMMON_FLAGS	:= -g -O3 -mrvl -Wall $(MACHDEP)
-INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I$(LIBOGC_INC)/SDL
+INCLUDES		:= -Iinclude -I$(LIBOGC_INC) -I$(INSTALL_HEADER_DIR)/SDL
 # -include sys/wait.h 
 # -include sys/syslimits.h
 
Index: smpeg/Makefile
===================================================================
--- smpeg/Makefile	(revision 99)
+++ smpeg/Makefile	(working copy)
@@ -89,7 +89,7 @@
 export INCLUDE	:=	$(foreach dir,$(INCLUDES), -iquote $(CURDIR)/$(dir)) \
 					$(foreach dir,$(LIBDIRS),-I$(dir)/include) \
 					-I$(CURDIR)/$(BUILD) \
-					-I$(LIBOGC_INC) -I$(LIBOGC_INC)/SDL
+					-I$(LIBOGC_INC) -I$(INSTALL_HEADER_DIR)/SDL
 
 #---------------------------------------------------------------------------------
 # build a list of library paths
@@ -115,9 +115,9 @@
 	
 install: $(BUILD)
 	@mkdir -p $(INSTALL_HEADER_DIR) $(INSTALL_LIB_DIR)
-	@mkdir -p $(INSTALL_HEADER_DIR)/mpeg
+	@mkdir -p $(INSTALL_HEADER_DIR)/smpeg
 	@cp -frv $(LIB_DIR)/*.* $(INSTALL_LIB_DIR)
-	@cp -frv include/*.* $(INSTALL_HEADER_DIR)/mpeg
+	@cp -frv include/*.* $(INSTALL_HEADER_DIR)/smpeg
 
 #---------------------------------------------------------------------------------
 run:
